oauth2/src/main/java/com/okta/spring/oauth/http/UserAgent.java

```diff
@@ -0,0 +1,485 @@
+
+    //Web Servers
+    private static final String WEB_SERVER_TOMCAT_ID = "tomcat";
+    private static final String WEB_SERVER_TOMCAT_BOOTSTRAP_CLASS = "org.apache.catalina.startup.Bootstrap";
+    private static final String WEB_SERVER_TOMCAT_EMBEDDED_CLASS = "org.apache.catalina.startup.Tomcat";
+    private static final String WEB_SERVER_JETTY_ID = "jetty";
+    private static final String WEB_SERVER_JETTY_CLASS = "org.eclipse.jetty.servlet.listener.ELContextCleaner";
+    private static final String WEB_SERVER_JBOSS_ID = "jboss";
+    private static final String WEB_SERVER_JBOSS_CLASS = "org.jboss.as.security.plugins.AuthenticationCacheEvictionListener";
+    private static final String WEB_SERVER_WEBSPHERE_ID = "websphere";
+    private static final String WEB_SERVER_WEBSPHERE_CLASS = "com.ibm.websphere.product.VersionInfo";
+    private static final String WEB_SERVER_GLASSFISH_ID = "glassfish";
+    private static final String WEB_SERVER_GLASSFISH_CLASS = "com.sun.enterprise.glassfish.bootstrap.GlassFishMain";
+    private static final String WEB_SERVER_WEBLOGIC_ID = "weblogic";
+    private static final String WEB_SERVER_WEBLOGIC_CLASS = "weblogic.version";
+    private static final String WEB_SERVER_WILDFLY_ID = "wildfly";
+    private static final String WEB_SERVER_WILDFLY_CLASS = "org.jboss.as.security.ModuleName";
+
+    private static final String VERSION_SEPARATOR = "/";
+    private static final String ENTRY_SEPARATOR = " ";
+
+    //Placeholder for the actual User-Agent String
+    private static final String USER_AGENT = createUserAgentString();
+
+    private UserAgent() {
+    }
+
+    public static String getUserAgentString() {
+        return USER_AGENT;
+    }
+
+    private static String createUserAgentString() {
+        String userAgent =
+                getOktaSpringString() +                 // okta-spring-security
+                getOktaShiroString() +                  // okta-shiro
+                getOktaSDKComponentsString() +          // okta-servlet-java | okta-spring-boot-starter
+                getOktaSdkString() +                    // okta-sdk-java
+                getSecurityFrameworkString() +          // shiro | spring-security
+                getIntegrationRuntimeString() +         // spring
+                getSpringBootString() +                 // spring-boot
+                getWebServerString() +                  // tomcat | jetty | jboss | websphere | glassfish | weblogic | wildfly
+                getJavaSDKRuntimeString() +             // java
+                getOSString();                          // Mac OS X
+        return userAgent.trim();
+    }
+
+    private static String getOktaShiroString() {
+        String integrationString;
+        integrationString = getFullEntryStringUsingPomProperties(INTEGRATION_SHIRO_CLASS, INTEGRATION_SHIRO_ID);
+        if(StringUtils.hasText(integrationString)) {
```

Comment:
nit: Consistency throughout this PR on space or no space after `if`

Reply:
+1